src/esssans/loki/general.py

```diff
@@ -0,0 +1,69 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+"""
+Default parameters, providers and utility functions for the loki workflow.
+"""
+
+
+from ..types import (
+    DataWithLogicalDims,
+    DetectorPixelShape,
+    Incident,
+    LabFrameTransform,
+    LoadedFileContents,
+    NexusDetectorName,
+    NexusInstrumentPath,
+    NeXusMonitorName,
+    NexusSourceName,
+    RawData,
+    RunType,
+    TransformationChainPath,
+    Transmission,
+)
+
+default_parameters = {
+    NexusInstrumentPath: 'instrument',
+    NexusDetectorName: 'larmor_detector',
+    NeXusMonitorName[Incident]: 'monitor_1',
+    NeXusMonitorName[Transmission]: 'monitor_2',
+    NexusSourceName: 'source',
+    # TODO: sample is not in the files, so by not adding the name here, we use the
+    # default value of [0, 0, 0] when loading the sample position.
+    TransformationChainPath: 'transformation_chain',
+}
+
+
+def to_logical_dims(da: RawData[RunType]) -> DataWithLogicalDims[RunType]:
+    return DataWithLogicalDims[RunType](
+        da.fold(
+            dim='detector_number', sizes=dict(layer=4, tube=32, straw=7, pixel=512)
+        ).flatten(dims=['tube', 'straw'], to='straw')
+    )
+
+
+def detector_pixel_shape(
+    dg: LoadedFileContents[RunType],
+    instrument_path: NexusInstrumentPath,
+    detector_name: NexusDetectorName,
+) -> DetectorPixelShape[RunType]:
+    return DetectorPixelShape[RunType](
+        dg[instrument_path][detector_name]['pixel_shape']
+    )
+
+
+def detector_lab_frame_transform(
+    dg: LoadedFileContents[RunType],
+    instrument_path: NexusInstrumentPath,
+    detector_name: NexusDetectorName,
+    transform_path: TransformationChainPath,
+) -> LabFrameTransform[RunType]:
+    return LabFrameTransform[RunType](
+        dg[instrument_path][detector_name][transform_path]
```

Comment:
Do you have a plan for handling multiple detector banks, where all this will be indepdendent?

Reply:
No, I didn't think of that...
Would you then have a `DetectorBank` parameter like the `RunType` and have `DataWithLogicalDims[DetectorBank, RunType]` everywhere?
Or should we just simply handle this in the loader, inserting required info in the data groups such as bank logical shape and pixel shape?

I guess merging all the contributions to `Q` at the end would also need to merge all the banks...

Reply:
Right now, I'm thinking of using a parameter table for the different banks, and then merging all the contributions to $I(Q)$ just before the normalization.

Reply:
Conclusion is: we leave it as is for now, and we will think about handling multiple bands later.